examples/ring_app/README.md

```diff
@@ -0,0 +1,152 @@
+# Simple Ring App with Authorization Example
+
+This example demonstrates how to incorporate the trapperkeeper-authorization
+service into a simple Ring app.  This is based loosely upon the
+[ring_app example] (https://github.com/puppetlabs/trapperkeeper-webserver-jetty9/tree/master/examples/ring_app)
+in the trapperkeeper-webservices-jetty9 project.  See that example for more
+information on the use of the `jetty9-service` and the Jetty web server
+integration with Ring.
+
+This example includes a simple set of default authorization rules which can be
+used to protect access to the `/hello` endpoint that is handled by the service.
+
+All code needed to execute this example is located in `./src/examples/ring_app`.
+The Clojure code is contained in the `ring_app.clj` file.
+
+## Launching trapperkeeper and running the app
+
+To start up _trapperkeeper_ and launch the sample application, use the
+following _lein_ command while in the _trapperkeeper-authorization_ home
+directory:
+
+~~~~sh
+lein trampoline run --config ./examples/ring_app/ring-example.conf \
+                    --bootstrap-config ./examples/ring_app/bootstrap.cfg
+~~~~
+
+### The `bootstrap.cfg` file
+
+The bootstrap config file contains a list of services that _trapperkeeper_ will
+load up and make available.  They are listed as fully-qualified Clojure
+namespaces and service names. For this example, the bootstrap.cfg looks like
+this:
+
+~~~~
+puppetlabs.trapperkeeper.services.webserver.jetty9-service/jetty9-service
+puppetlabs.trapperkeeper.services.authorization.authorization-service/authorization-service
+examples.ring-app.ring-app/hello-service
+~~~~
+
+This configuration indicates that the jetty9 `WebserverService`, authorization
+service, and new "hello" service, defined in the `ring_app.clj` file, are to be
+loaded.
+
+### The `ring-example.conf` configuration file
+
+For the application configuration, a file called `ring-example.conf` provides
+a fairly minimal configuration for the hello service:
+
+~~~~hocon
+global: {
+    logging-config: ./examples/ring_app/logback.xml
+}
+
+webserver: {
+    port: 8080
+    ssl-port: 8081
+    ssl-cert: "./dev-resources/config/jetty/ssl/certs/localhost.pem"
+    ssl-ca-cert: "./dev-resources/config/jetty/ssl/certs/ca.pem"
+    ssl-key: "./dev-resources/config/jetty/ssl/private_keys/localhost.pem"
```

Comment:
What do you think about putting these SSL files under the examples/ring_app to keep it all self-contained under one directory?